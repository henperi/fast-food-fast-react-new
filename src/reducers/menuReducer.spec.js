/* eslint-disable no-undef */
import menuReducers from './menuReducer';

describe('menuReducers', () => {
  const initialState = [];
  it('should return the default state if no action is specified', () => {
    const action = { type: '' };
    expect(menuReducers(initialState, action)).toMatchSnapshot({});
    expect(menuReducers(initialState, action)).toEqual(initialState);
  });

  it('should update the state and SET_MENU when type is SET_MENU and payload is specified', () => {
    const action = {
      type: 'SET_MENU',
      payload: [{ food_id: 5, name: 500 }],
    };
    expect(menuReducers(initialState, action)).toMatchSnapshot({});
    expect(menuReducers(initialState, action)).toEqual({ 0: { food_id: 5, name: 500 } });
  });
});
