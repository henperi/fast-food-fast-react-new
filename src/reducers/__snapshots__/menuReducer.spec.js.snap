// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menuReducers should return the default state if no action is specified 1`] = `Object {}`;

exports[`menuReducers should update the state and SET_MENU when type is SET_MENU and payload is specified 1`] = `
Object {
  "0": Object {
    "food_id": 5,
    "name": 500,
  },
}
`;
